lib/elixir/lib/kernel.ex

```diff
@@ -3711,6 +3711,17 @@ defmodule Kernel do
       target = Keyword.get(opts, :to) ||
         raise ArgumentError, "expected to: to be given as argument"
 
+      if is_list(funs) do
+        IO.write :stderr, "warning: passing a list to Kernel.defdelegate/2 is deprecated, " <>
```

Comment:
We should always use `:elixir_errors.warn` in compile time as it will be tracked by compiler and affect `warnings_as_errors` option.
As you see many of those examples in search result will be called in runtime – `IO.write` is the choice then.